From ac198b30257075166e522fad611fcc456a26f003 Mon Sep 17 00:00:00 2001
From: Vinita Gupta <vinita.gupta@caviumnetworks.com>
Date: Tue, 16 Oct 2018 12:52:40 +0300
Subject: [PATCH 0278/1051] octeontx-pki: Enable/fix the pki packet drop on
 port open

Signed-off-by: Vinita Gupta <vinita.gupta@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c b/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
index d3f24fc0c188..da0004d79a04 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
@@ -71,7 +71,9 @@ static void reset_port_reg(struct pki_t *pki, struct pki_port *port)
 			      cfg);
 		pki_reg_write(pki, PKI_CLX_PKINDX_LG_CUSTOM(i, port->pkind),
 			      cfg);
+		cfg = 0x1ull << PKI_STYLE_CFG_DROP_SHIFT;
 		pki_reg_write(pki, PKI_CLX_STYLEX_CFG(i, style), cfg);
+		cfg = 0x0;
 		pki_reg_write(pki, PKI_CLX_STYLEX_CFG2(i, style), cfg);
 		pki_reg_write(pki, PKI_CLX_STYLEX_ALG(i, style), cfg);
 	}
@@ -196,6 +198,7 @@ int pki_port_open(struct pkipf_vf *vf, u16 vf_id,
 		cfg |= (0x1ULL << PKI_STYLE_CFG_FCS_STRIP_SHIFT);
 	}
 	cfg |= (0x1ULL << PKI_STYLE_CFG_LENERR_EN_SHIFT);
+	cfg |= (0x1ull << PKI_STYLE_CFG_DROP_SHIFT);
 	for (i = 0; i < pki->max_cls; i++)
 		pki_reg_write(pki, PKI_CLX_STYLEX_CFG(i, port->init_style),
 			      cfg);
-- 
2.17.1

